test/node/src/safe-empty-where-in.test.ts

```diff
@@ -0,0 +1,916 @@
+            ],
+            parameters: ['John', 'Mary'],
+          },
+          sqlite: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" not in (?, ?)`,
+            ],
+            parameters: ['John', 'Mary'],
+          },
+        })
+
+        result = await notInQuery.execute()
+
+        expect(result).to.have.length(1)
+        expect(result).to.deep.equal([{ firstName: 'Tom' }])
+      })
+    })
+
+    describe('DELETE FROM WHERE IN TESTS', () => {
+      it('should handle deleteFrom without returning with no runtime errors', async () => {
+        const query = db
+          .deleteFrom('safeEmptyArrayPerson')
+          .where('firstName', 'in', [])
+
+        testSql(query, dialect, {
+          postgres: {
+            sql: [
+              `delete from "safeEmptyArrayPerson"`,
+              `where "firstName" in ($1)`,
+            ],
+            parameters: [null],
+          },
+          mysql: {
+            sql: [
+              'delete from `safeEmptyArrayPerson`',
+              'where `firstName` in (?)',
+            ],
+            parameters: [null],
+          },
+          mssql: {
+            sql: [
+              `delete from "safeEmptyArrayPerson"`,
+              `where "firstName" in (@1)`,
+            ],
+            parameters: [null],
+          },
+          sqlite: {
+            sql: [
+              `delete from "safeEmptyArrayPerson"`,
+              `where "firstName" in (?)`,
+            ],
+            parameters: [null],
+          },
+        })
+
+        let result = await query.execute()
+
+        expect(result).to.deep.equal([new DeleteResult(BigInt(0))])
+
+        const notInQuery = db
+          .deleteFrom('safeEmptyArrayPerson')
+          .where('firstName', 'not in', [])
+
+        testSql(notInQuery, dialect, {
+          postgres: {
+            sql: [
+              `delete from "safeEmptyArrayPerson"`,
+              `where "firstName" not in ($1)`,
+            ],
+            parameters: [null],
+          },
+          mysql: {
+            sql: [
+              'delete from `safeEmptyArrayPerson`',
+              'where `firstName` not in (?)',
+            ],
+            parameters: [null],
+          },
+          mssql: {
+            sql: [
+              `delete from "safeEmptyArrayPerson"`,
+              `where "firstName" not in (@1)`,
+            ],
+            parameters: [null],
+          },
+          sqlite: {
+            sql: [
+              `delete from "safeEmptyArrayPerson"`,
+              `where "firstName" not in (?)`,
+            ],
+            parameters: [null],
+          },
+        })
+
+        result = await notInQuery.execute()
+
+        expect(result).to.deep.equal([new DeleteResult(BigInt(0))])
+      })
+
+      it('should handle deleteFrom with returning in supported dialects', async () => {
```

Comment:
irrelevant. remove.

Reply:
Will this use case never be used?

Reply:
Testing every possible API combination for every plugin that ever alters the AST is pointless.

Reply:
okay, will defer to your opinion on this. Deleted.